server/service/src/requisition/request_requisition/insert_program.rs

```diff
@@ -63,26 +66,66 @@ pub fn insert_program_request_requisition(
 }
 
 fn validate(
-    connection: &StorageConnection,
-    store_id: &str,
+    ctx: &ServiceContext,
     input: &InsertProgramRequestRequisition,
-) -> Result<(), OutError> {
+) -> Result<(ProgramRow, ProgramRequisitionOrderTypeRow), OutError> {
+    let connection = &ctx.connection;
+
     if let Some(_) = check_requisition_exists(connection, &input.id)? {
         return Err(OutError::RequisitionAlreadyExists);
     }
 
-    // TODO: Check if we already have reached out max requisitions for this period
-    // https://github.com/openmsupply/open-msupply/issues/1599
+    let requisitions = RequisitionRepository::new(&connection).query_by_filter(
+        RequisitionFilter::new().order_type(EqualFilter::equal_to(&input.program_order_type_id)),
+    )?;
 
-    // TODO: Check Order Type is valid for this Program
-    // https://github.com/openmsupply/open-msupply/issues/1599
+    let program_settings = get_program_requisition_settings(ctx, &ctx.store_id)?;
 
-    // TODO: Check if the 'other_party' is a valid Program Supplier
-    // https://github.com/openmsupply/open-msupply/issues/1599
+    let order_type: ProgramRequisitionOrderTypeRow = program_settings
+        .iter()
+        .map(|setting| {
+            setting
+                .order_types
+                .iter()
+                .filter(|order_type| order_type.order_type.id == input.program_order_type_id)
+                .map(|order_type| order_type.order_type.clone())
+        })
+        .flatten()
+        .next()
+        .ok_or(OutError::ProgramOrderTypeDoesNotExist)?;
+
+    if requisitions.len() as i32 >= order_type.max_order_per_period {
```

Comment:
we already do the count of requisitions vs max_order_per_period in get_program_requisition, availablePeriods on `orderType`:

https://github.com/openmsupply/open-msupply/blob/e47cfc42412dabc7a0cf77cf236262fe6b9c7304/server/service/src/requisition/program_settings/map.rs#L31-L41

https://github.com/openmsupply/open-msupply/blob/e47cfc42412dabc7a0cf77cf236262fe6b9c7304/server/service/src/requisition/program_settings/map.rs#L98

https://github.com/openmsupply/open-msupply/blob/e47cfc42412dabc7a0cf77cf236262fe6b9c7304/server/repository/src/migrations/v1_01_11/requisition.rs#L20-L22

Reply:
[7fcebcb](https://github.com/openmsupply/open-msupply/pull/1660/commits/7fcebcbda2462bad53ba428bdf354bf61cb3ad34) thanks Andrei~